superset/connectors/sqla/models.py

```diff
@@ -877,6 +878,29 @@ def _get_sqla_row_level_filters(
                 _("Error in jinja expression in RLS filters: %(msg)s", msg=ex.message,)
             )
 
+    def _get_template_kwargs(  # pylint:disable = too-many-arguments
+        self,
+        metrics: List[Metric],
+        extra_cache_keys: List[Any],
+        filter_: Optional[List[Dict[str, Any]]] = None,
+        from_dttm: Optional[datetime] = None,
+        to_dttm: Optional[datetime] = None,
+        groupby: Optional[List[str]] = None,
+        row_limit: Optional[int] = None,
+        row_offset: Optional[int] = None,
+    ) -> Dict[str, Any]:
+        return {
+            "from_dttm": from_dttm.isoformat() if from_dttm else None,
+            "groupby": groupby,
+            "metrics": metrics,
+            "row_limit": row_limit,
+            "row_offset": row_offset,
+            "to_dttm": to_dttm.isoformat() if to_dttm else None,
+            "filter": filter_,
+            "columns": [col.column_name for col in self.columns],
+            "extra_cache_keys": extra_cache_keys,
+        }
+
```

Comment:
Would it make sense to combine `_get_template_kwargs` and `update_template_kwargs` into one function in `sql_query_utils.py`?